src/main/surfrdMod.F90

```diff
@@ -658,12 +787,12 @@ subroutine surfrd_veg_all(begg, endg, ncid, ns, actual_numcft)
           if ( use_fates ) then
              if ( masterproc ) write(iulog,*) "WARNING: When fates is on we allow new CFT based surface datasets ", &
                                               "to be used with create_crop_land FALSE"
-             cftsize = 2
-             allocate(array2DCFT (begg:endg,cft_lb:cftsize-1+cft_lb))
-             allocate(array2DFERT(begg:endg,cft_lb:cftsize-1+cft_lb))
-             call surfrd_cftformat( ncid, begg, endg, array2DCFT, array2DFERT, cftsize, natpft_size-cftsize ) ! Read crops in as CFT's
```

Comment:
This is the part that illustrates why I think surfrd_wtfates should be refactored. The previous setup called a shared function surfrd_cftformat to do the read, and then a small (unit-testable) function convert_cft_to_pft to do the conversion required for create_crop_landunit=FALSE. When FATES could handle create_crop_landunit=TRUE, I envisioned that second function would go away.

The problem with surfrd_wtfates is that it duplicates (but subtly modifies) surfrd_cftformat and also moves an equivalent of convert_cft_to_pft to inside of surfrd_wtfates inside of it, so it's no longer unit-testable. It also means the most obvious way to add the ability to handle different scenarios with FATES in regard to crop landunits is to create duplicated versions of wrfrd_wtfates all subtly different.

From your work I can see that the previous paradigm of 

surfrd_cftformat
convert_cft_to_pft

will not work, but I think the following would:

surfrd_cftformat
if ( .not. use_fates_sp) fill_fates_pftwts
convert_cft_to_pft

convert_cft_to_pft could be able to be changed directly to your new formulation as it's only used there. surfrd_cftformat might need some slight changes to work with FATES, but not much.

Reply:
What that allows us to do is handle a few different conditions for example to change to the following which I think encapsulates all the different options we might want to have...

``` FORTRAN
if ( use_fates )
  call surfrd_cftformat
  if ( .not. use_fates_sp) call fill_fates_pftwts
  if ( create_crop_landunit ) then
     if ( fates_crops_on_croplu )then
        call convert_cft_to_fates_cft
     else
        call convert_cft_to_pft_v1
    end if
  end if
end if
```

The convert_cft_to_pft could have the two flavors you provided and when answer changes are allowed we move to the "_v2" flavor.

Reply:
This is the part that I'll refactor in a future PR.

Reply:
I want to keep this one open so it can be seen for the future work. But, it can be considered closed for this PR.